### PR TITLE
Add status section for plans and authentication services

### PR DESCRIPTION
**Pull request by lulf**
### Type of change

- Bugfix
- Enhancement / new feature

### Description

Add status field to authentication service, address space plan and address plan resources. Update status based on validation/reconciliation process.

Infra configs are not in this list because there is no validation beyond openapi provided at present.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by lulf** [comment]
@k-wall I realize I need to do some additional fixes here to ensure other users of KubeSchemaApi do not conflict with the one in address-space-controller

**Comment by lulf** [comment]
This is how it looks now:

```
[lulf@pteppic enmasse]$ oc get addressspaceplan standard-foo -o wide
NAME           PHASE     STATUS                                                                                                                              AGE
standard-foo   Pending   Error validating address space plan standard-foo: Error validating address space plan standard-foo: missing resources [aggregate]   30s
[lulf@pteppic enmasse]$ oc edit addressspaceplan standard-foo 
addressspaceplan.admin.enmasse.io/standard-foo edited
[lulf@pteppic enmasse]$ oc get addressspaceplan standard-foo -o wide
NAME           PHASE    STATUS   AGE
standard-foo   Active            38s

```

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated

File: templates/address-space-controller/020-Role-address-space-controller.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ metadata:
 rules:
   - apiGroups: [ "admin.enmasse.io" ]
     resources: [ "addressplans", "addressspaceplans", "brokeredinfraconfigs", "standardinfraconfigs", "authenticationservices", "consoleservices"]
-    verbs: [ "get", "list", "watch" ]
+    verbs: [ "create", "update", "patch", "get", "list", "watch", "delete" ]
```
Comment (k-wall) [inline templates/address-space-controller/020-Role-address-space-controller.yaml]: Why the need to delete, create and update these resources?  I see this change requires patch ability for the address-space-controller on the plans, but I'm not sure why the other rights are required.

Comment (lulf) [reply templates/address-space-controller/020-Role-address-space-controller.yaml]: I admit it was just a copy paste from other rights. I will reduce it patch only.

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated

**Comment by k-wall** [comment]
There is a defect that means the address status message is duplicated.  The defect manifests if a address plan is used by more than one address space plan.  The issue is addressSpacePlan loop in assembleSchema.  ` addressPlanStatusMap.get(addressPlanName).appendMessage(e.getMessage());` has the effect of duplicating the message.

status:
  message: 'Error validating address plan standard-small-queue: missing resources
    [router]. Error validating address plan standard-small-queue: missing resources
    [router]. Error validating address plan standard-small-queue: missing resources
    [router]. Error validating address plan standard-small-queue: missing resources
    [router]'
  phase: Pending

**Comment by k-wall** [comment]
I also notice that  address plan `spec.addressType` is not validated.   For instance I can defined an address plan with address type muticast and give it to a brokered addressspace plan.

**Comment by lulf** [comment]
@k-wall Thanks. I've added a somewhat hacky fix, but it was also a lot simpler than the alternative.

**Comment by k-wall** [comment]
> @k-wall Thanks. I've added a somewhat hacky fix for the duplicated message, but it was also a lot simpler than the alternative.

I was wondering about making `Messages []string` as we already have elsewhere.   I'm not too bothered either way, so if you prefer it the way it is, that okay with me.

**Comment by lulf** [comment]
@k-wall I'd like to make it string to align with what iotproject is doing, and try to be more kubernetes-native (strings are handled better than arrays in outputs for CRDs etc.)

**Comment by lulf** [comment]
@k-wall I've added address type validation, and added the Failed phase and set that if failing validation, making the issue a bit clearer.

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated

**Pull request merged**